### PR TITLE
rambox: 0.7.9 -> 2.0.9

### PR DESCRIPTION
**Pull request by nazarewk**
###### Description of changes

update rambox package, it does not prepare 32 bit images anymore

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
  - https://github.com/nazarewk-iac/nix-configs/commit/ed736c62
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/rambox/default.nix [hunk 1]
```diff
@@ -5,25 +5,21 @@ let
...
-      sha256 = "13wiciyshyrabq2mvnssl2d6svia1kdvwx3dl26249iyif96xxvq";
+      url = "https://github.com/ramboxapp/download/releases/download/v${version}/Rambox-${version}-linux-x64.AppImage";
+      sha256 = "sha256-o2ydZodmMAYeU0IiczKNlzY2hgTJbzyJWO/cZSTfAuM=";
     };
   }.${stdenv.system} or (throw "Unsupported system: ${stdenv.system}");
 
   meta = with lib; {
     description = "Free and Open Source messaging and emailing app that combines common web applications into one";
     homepage = "https://rambox.pro";
     license = licenses.mit;
```
Comment (bobby285271) [inline pkgs/applications/networking/instant-messengers/rambox/default.nix:20]:
```suggestion
    license = licenses.unfree;
```

File: pkgs/applications/networking/instant-messengers/rambox/default.nix [hunk 2]
```diff
@@ -5,25 +5,21 @@ let
...
-    maintainers = with maintainers; [];
-    platforms = [ "i686-linux" "x86_64-linux" ];
+    maintainers = with maintainers; [ nazarewk ];
+    platforms = [ "x86_64-linux" ];
     hydraPlatforms = [];
```
Comment (bobby285271) [inline pkgs/applications/networking/instant-messengers/rambox/default.nix:23]:
```suggestion
```

**Review by bobby285271 — COMMENTED** [review]
For some reasons I cannot find source code for this version, I guess it is now ... non-free?

Also maybe since the project is merged, can we remove [`rambox-pro`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/instant-messengers/rambox/pro.nix) at the same time? We can add an entry in `pkgs/top-level/aliases.nix` to alias it to this package. :thinking:

**Comment by nazarewk** [comment]
Thanks for review, had no idea about those

File: pkgs/applications/networking/instant-messengers/rambox/default.nix [hunk 3]
```diff
@@ -1,29 +1,43 @@
...
+{ appimageTools, lib, fetchurl, makeDesktopItem }:
 
 let
-  mkRambox = opts: callPackage (import ./rambox.nix opts) {};
-in
-mkRambox rec {
   pname = "rambox";
-  version = "0.7.9";
+  version = "2.0.9";
+  name = "${pname}-${version}";
```
Comment (bobby285271) [inline pkgs/applications/networking/instant-messengers/rambox/default.nix:6]:
```suggestion
```

File: pkgs/applications/networking/instant-messengers/rambox/default.nix [hunk 4]
```diff
@@ -1,29 +1,43 @@
...
+  desktopItem = (makeDesktopItem {
+    desktopName = "Rambox";
+    name = pname;
+    exec = pname;
+    icon = pname;
+    categories = [ "Network" ];
+  });
+
+  appimageContents = appimageTools.extractType2 {
+    inherit name src;
```
Comment (bobby285271) [inline pkgs/applications/networking/instant-messengers/rambox/default.nix:22]:
```suggestion
    inherit pname version src;
```

File: pkgs/applications/networking/instant-messengers/rambox/default.nix [hunk 5]
```diff
@@ -1,29 +1,43 @@
...
+  };
+in
+appimageTools.wrapType2 {
+  inherit name src;
```
Comment (bobby285271) [inline pkgs/applications/networking/instant-messengers/rambox/default.nix:26]:
```suggestion
  inherit pname version src;
```

**Review by bobby285271 — COMMENTED** [review]
Not really related to this bump, but I think we can use pname + version instead of name today (https://github.com/NixOS/nixpkgs/commit/db44b28a061fa44654dda12a9d862d4fdeecefa0), lets try :-)

File: pkgs/applications/networking/instant-messengers/rambox/default.nix [hunk 6]
```diff
@@ -1,29 +1,43 @@
...
-      sha256 = "19y4cmrfp79dr4hgl698imp4f3l1nhgvhh76j5laxg46ld71knil";
-    };
-    i686-linux = fetchurl {
-      url = "https://github.com/ramboxapp/community-edition/releases/download/${version}/Rambox-${version}-linux-i386.AppImage";
-      sha256 = "13wiciyshyrabq2mvnssl2d6svia1kdvwx3dl26249iyif96xxvq";
-    };
-  }.${stdenv.system} or (throw "Unsupported system: ${stdenv.system}");
+  extraInstallCommands = ''
+    mkdir -p $out/share/applications $out/share/icons/hicolor/256x256/apps
+    # CE uses rambox-<version>, Pro uses rambox
```
Comment (bobby285271) [inline pkgs/applications/networking/instant-messengers/rambox/default.nix:30]: Also I guess we don't need this comment anymore :-)

```suggestion
```

**Comment by nazarewk** [comment]
cleaned it up

**Comment by nazarewk** [comment]
yeah, totally missed it

File: pkgs/applications/networking/instant-messengers/rambox/default.nix [hunk 7]
```diff
@@ -1,29 +1,41 @@
...
+  '';
 
   meta = with lib; {
-    description = "Free and Open Source messaging and emailing app that combines common web applications into one";
-    homepage = "https://rambox.pro";
-    license = licenses.mit;
-    maintainers = with maintainers; [];
-    platforms = [ "i686-linux" "x86_64-linux" ];
-    hydraPlatforms = [];
+    description = "Workspace Simplifier - a cross-platform application organizing web services into Workspaces similar to browser profiles.";
```
Comment (bobby285271) [inline pkgs/applications/networking/instant-messengers/rambox/default.nix:35]: [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) says no period at the end :upside_down_face: 

```suggestion
    description = "Workspace Simplifier - a cross-platform application organizing web services into Workspaces similar to browser profiles";
```

File: pkgs/top-level/aliases.nix [hunk 8]
```diff
@@ -1267,6 +1267,7 @@ mapAliases ({
 
   radare2-cutter = cutter; # Added 2021-03-30
   railcar = throw "'railcar' has been removed, as the upstream project has been abandoned"; # Added 2022-06-27
+  rambox-pro = rambox; # Added 2021-03-30
```
Comment (bobby285271) [inline pkgs/top-level/aliases.nix:1270]:
```suggestion
  rambox-pro = rambox; # Added 2022-12-12
```

**Review by bobby285271 — COMMENTED** [review]
(Sorry for giving so many rounds of review, hopefully this is the last round... :upside_down_face:)

**Comment by bobby285271** [comment]
Thanks!

**Pull request merged**